src/Domain/Event/JSONDomainEventFactory.php

```diff
@@ -1,19 +1,22 @@
 <?php
 
-namespace Cmp\DomainEvent\Domain\Event;
-
-use Cmp\DomainEvent\Domain\Event\Exception\InvalidJSONDomainEventException;
-use Cmp\Queue\Domain\JSONMessageFactory;
+namespace Domain\Event;
 
+use Domain\Event\Exception\InvalidJSONDomainEventException;
+use Domain\Queue\JSONMessageFactory;
 
 class JSONDomainEventFactory implements JSONMessageFactory
 {
-
+    /**
+     * @param $json
+     * @return DomainEvent
+     * @throws InvalidJSONDomainEventException
+     */
     public function create($json)
     {
         try {
             $domainEventArray = json_decode($json, true);
-            return new DomainEvent($domainEventArray['origin'], $domainEventArray['name'], $domainEventArray['ocurredOn'], $domainEventArray['body']);
+            return new DomainEvent($domainEventArray['origin'], $domainEventArray['name'], $domainEventArray['occurredOn'], $domainEventArray['body']);
```

Comment:
What happens if the JSON does not contain the expected keys or it is in a bad format?